### PR TITLE
Project summary

### PR DESCRIPTION
**Pull request by ywrsusan**
Jira Issue
https://jira4.prod.bloomberg.com/browse/RDISCROWD-1575

Describe your changes
created a summary page to show necessary configurations current project has, with redirect links to original configuration pages for editing purpose.

### CODE REVIEW

File: static/src/components/fieldsconfig/categorical_field_config.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@
         :name="name"
         :type="type"
         :retry-for-consensus="retryForConsensus"
-        :editable="true"
+        :is-editable="true"
```
Comment (mariarivera30) [inline static/src/components/fieldsconfig/categorical_field_config.vue]: I don't see this prop `is-editable` on this component `static/src/components/fieldsconfig/field_config_base.vue`

File: static/src/components/fieldsconfig/consensus_config.vue [hunk 2]
```diff
@@ -1,78 +1,58 @@
...
   >
-    <div class="col-md-12">
-      <h3> Consensus Configuration </h3>
-      <div class="form-inline ">
-        <p> Consensus threshold: </p>
-        <div class="input-group">
+    <div
+      v-if="hasRetryFields"
+      class="col-md-12"
+      style="width:85%"
```
Comment (mariarivera30) [inline static/src/components/fieldsconfig/consensus_config.vue]: Can we create a class and move this style out

File: static/src/components/fieldsconfig/consensus_config.vue [hunk 3]
```diff
@@ -1,78 +1,58 @@
...
-            <p> Add redundancy to retry</p>
-            <p> Maximum redundancy</p>
-          </div>
-          <div class="col-md-3 align-right">
-            <p> {{ consensusThreshold }}</p>
-            <p> {{ redundancyConfig }}</p>
-            <p> {{ maxRetries }}</p>
-          </div>
+      <div class="form-group row">
+        <div class="col-sm-4">
```
Comment (mariarivera30) [inline static/src/components/fieldsconfig/consensus_config.vue]: Not big deal but I think we should be consistent with `col-sm`, Ill prefer md

File: static/src/components/setting/task_setting.vue [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+      timeoutMinute: Math.floor(this.taskTimeout / 60),
+      timeoutSecond: this.taskTimeout % 60,
+      defaultRedundancy: this.taskRedundancy,
+      currentRedundancy: null
+    };
+  },
+
+  methods: {
+    getOptions () {
+      var options = [];
```
Comment (mariarivera30) [inline static/src/components/setting/task_setting.vue]: Can we use let instead of var?

File: static/src/components/setting/index.vue [hunk 5]
```diff
@@ -0,0 +1,326 @@
+<template>
+  <div class="stats-config row">
+    <div class="col-sm-12">
+      <div class="form-group row">
+        <div class="col-sm-4">
+          <p> bcos bucket </p>
```
Comment (fiorda) [inline static/src/components/setting/index.vue]: what's a `bcos`?

File: static/src/components/setting/index.vue [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+              class="form-check-label"
+              for="defaultCheck1"
+            >
+              <p> L4: Crowdsourcing Worker and users in L3 access  </p>
+            </label>
+          </div>
+        </div>
+      </div>
+      <div
+        v-if="isPrivate"
```
Comment (fiorda) [inline static/src/components/setting/index.vue]: what is private?

File: static/src/components/setting/index.vue [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+      L2: false,
+      L3: false,
+      L4: false,
+      search: '',
+      searchResult: this.users
+    };
+  },
+
+  mounted: function () {
+    var _this = this;
```
Comment (fiorda) [inline static/src/components/setting/index.vue]: why was `_this` variable introduced?

File: static/src/components/setting/index.vue [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+      default: () => ({ gigwork_poller: { target_bucket: null } })
+    }
+  },
+
+  data () {
+    return {
+      bcos: this.externalConfig.gigwork_poller.target_bucket,
+      assignee: {},
+      users: {},
+      L1: false,
```
Comment (fiorda) [inline static/src/components/setting/index.vue]: currently levels are defined in the server's configuration file, not hard-coded

File: static/src/components/setting/index.vue [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+      for (var i = 0; i < this.dataAccess.length; i++) {
+        if (this.dataAccess[i] === level) {
+          return true;
+        }
+      }
+      return false;
+    },
+
+    getAssignedUsers () {
+      var users = {};
```
Comment (fiorda) [inline static/src/components/setting/index.vue]: can we use `let` or `const`?

File: static/src/components/setting/index.vue [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+    _this.searchResult = _this.users;
+    _this.L1 = _this.hasLevel('L1');
+    _this.L2 = _this.hasLevel('L2');
+    _this.L3 = _this.hasLevel('L3');
+    _this.L4 = _this.hasLevel('L4');
+  },
+
+  methods: {
+    hasLevel (level) {
+      for (var i = 0; i < this.dataAccess.length; i++) {
```
Comment (fiorda) [inline static/src/components/setting/index.vue]: can we use `this.dataAccess.some` ?

File: static/src/components/setting/index.vue [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+      var users = {};
+      for (var i = 0; i < this.assignUsers.length; i++) {
+        users[this.assignUsers[i].id] = this.assignUsers[i];
+      }
+      return users;
+    },
+
+    getUsers () {
+      var users = {};
+      for (var i = 0; i < this.allUsers.length; i++) {
```
Comment (fiorda) [inline static/src/components/setting/index.vue]: can we use `this.allUsers.forEach`?

File: static/src/components/setting/ownership_setting.vue [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+    return {
+      coowners: {},
+      searchResult: [],
+      search: ''
+
+    };
+  },
+
+  mounted: function () {
+    var _this = this;
```
Comment (fiorda) [inline static/src/components/setting/ownership_setting.vue]: why is this necessary?

File: static/src/components/setting/ownership_setting.vue [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+    _this.coowners = _this.getCoowners();
+  },
+
+  methods: {
+
+    getCoowners () {
+      var users = {};
+      for (var i = 0; i < this.coOwners.length; i++) {
```
Comment (fiorda) [inline static/src/components/setting/ownership_setting.vue]: can we use `array.prototype`'s methods instead of a`for` loop? (similar below)

File: static/src/components/setting/ownership_setting.vue [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+          users[this.coOwners[i].id] = this.coOwners[i];
+      }
+      return users;
+    },
+
+    prepareResult () {
+      var found = this.searchResult;
+      this.searchResult = [];
+      for (var i = 0; i < found.length; i++) {
```
Comment (fiorda) [inline static/src/components/setting/ownership_setting.vue]: use `found.map`?

File: static/src/components/setting/index.vue (hunk 8 above)
Comment (ywrsusan) [reply static/src/components/setting/index.vue]: do you mean we should read access levels from configuration file in case there're changes on  levels?

Comment (fiorda) [reply static/src/components/setting/index.vue]: The code reads the levels from the configuration, so one can spin up a server with any access level they want. This page would only make sense for a particular choice of the levels (L1 L2 L3 L4)

Comment (ywrsusan) [reply static/src/components/setting/index.vue]: agree, will fix in next commit

File: static/src/components/setting/index.vue [hunk 15]
```diff
@@ -0,0 +1,387 @@
...
+                :value="choice[0]"
+              >
+                {{ choice[1] }}
+              </option>
+            </select>
+          </div>
+        </div>
+      </div>
+      <div
+        v-if="privateInstance"
```
Comment (fiorda) [inline static/src/components/setting/index.vue]: in the past we tried avoiding coding based on particular instances, rather we tried thinking in terms of features that a user can enable. In this case, would it make sense to make the block visible only when the levels are configured? (e.g., something like `v-if=isNonEmpty(accessLevels)`)

File: static/src/components/setting/index.vue [hunk 16]
```diff
@@ -0,0 +1,387 @@
...
+        <div class="col-md-5">
+          <p> assign users </p>
+        </div>
+        <div class="col-md-7 pull-right">
+          <table
+            class="table table-striped table-hover"
+            style="border: 1px ridge silver"
+          >
+            <thead>
+              <tr style="border-top: 1px ridge silver">
```
Comment (fiorda) [inline static/src/components/setting/index.vue]: Can we avoid inline style? Can part of this stile be made generic and reused for consistency across pages?

File: static/src/components/fieldsconfig/index.vue [hunk 17]
```diff
@@ -1,18 +1,31 @@
 <template>
   <div class="stats-config row">
-    <div class="col-md-12">
-      <h3> Answer Field Configuration </h3>
+    <div
+      class="col-md-12"
+      style="width:85%"
```
Comment (mariarivera30) [inline static/src/components/fieldsconfig/index.vue]: I think you missed this inline style.

File: static/src/components/setting/task_setting.vue [hunk 18]
```diff
@@ -0,0 +1,190 @@
...
+        <div class="col-md-6">
+          <p> timeout </p>
+        </div>
+        <div class="col-md-6 pull-right">
+          <div class="input-group">
+            <input
+              v-model="timeoutMinute"
+              type="text"
+              class="form-control input-sm"
+              style="width: 100%"
```
Comment (mariarivera30) [inline static/src/components/setting/task_setting.vue]: Why is this ` style="width: 100%"` added on every input?

File: static/src/components/setting/task_setting.vue [hunk 19]
```diff
@@ -0,0 +1,196 @@
...
+            </option>
+          </select>
+        </div>
+      </div>
+      <div class="form-group row">
+        <div class="col-md-6">
+          <p> randomize order with same priority </p>
+        </div>
+        <div class="col-md-6 pull-right">
+          <label class="switch">
```
Comment (fiorda) [inline static/src/components/setting/task_setting.vue]: This is relying on css defined on a different component. Perhaps it is convenient to move the switch style to a scss file so it can be reused across different pages.

File: static/src/components/setting/task_setting.vue [hunk 20]
```diff
@@ -0,0 +1,196 @@
...
+            <input
+              v-model="random"
+              type="checkbox"
+            >
+            <span class="slider" />
```
Comment (fiorda) [inline static/src/components/setting/task_setting.vue]: seems to also use style defined in another component?

File: static/src/components/setting/quiz_setting.vue [hunk 21]
```diff
@@ -0,0 +1,293 @@
...
+          window.pybossaNotify('An error occurred.', true, 'error');
+        }
+      } catch (error) {
+        window.pybossaNotify('An error occurred.', true, 'error');
+      }
+     }
+  }
+};
+
+</script>
```
Comment (fiorda) [inline static/src/components/setting/quiz_setting.vue]: I think it is a decent practice to scope a component style by default, see for example 

https://github.com/bloomberg/pybossa-default-theme/blob/fd0ed64026aa7237a0447f5b58647f65b4474fa3/static/src/components/builder/components/TextInput/TextInputForm.vue#L51-L57

File: static/src/components/setting/index.vue [hunk 22]
```diff
@@ -0,0 +1,365 @@
...
+  overflow-y: scroll;
+}
+.form-check-label p {
+  margin-right:10px
+}
+.title {
+  border-top: 1px ridge silver
+}
+#assign-user-column {
+  border-right: 1px ridge silver; width:50%
```
Comment (fiorda) [inline static/src/components/setting/index.vue]: in general there seems to be a lot of specific hardcoded style (colors, line widths, borders, margin, paddings). In my opinion this will make it harder to keep the style consistent, especially if at some point the global style is changed. Again my opinion it's better instead use scss, reuse already defined colors and styles if possible and create appropriate classes when needed.

File: static/src/components/setting/task_setting.vue (hunk 19 above)
Comment (fiorda) [reply static/src/components/setting/task_setting.vue]: Curious, why you decided to use switches instead of checkboxes?

File: static/src/components/setting/task_setting.vue (hunk 20 above)
Comment (mariarivera30) [reply static/src/components/setting/task_setting.vue]: Maybe when we merge Kory PR you can do what she is doing to have a file with all  the css.

Comment (fiorda) [reply static/src/components/setting/task_setting.vue]: scss can be used already

File: static/src/components/setting/index.vue (hunk 22 above)
Comment (ywrsusan) [reply static/src/components/setting/index.vue]: agree, already fixed in latest commit.

File: static/src/components/setting/task_setting.vue (hunk 19 above)
Comment (ywrsusan) [reply static/src/components/setting/task_setting.vue]: Agree, already move the style to a sass file. and it seems that "switch" is the official name for this UI component.
